File: java/systemparams/machine.java
import java.util.*;

// emitted from framec_v0.10.0
// get include files at https://github.com/frame-lang/frame-ancillary-files

enum SystemParamState {
    BEGIN(0), END(1);

    public final int value;

    private SystemParamState(int value) {
        this.value = value;
    }
    
    public int getValue() { return value; }
}

class SystemParams {

    private SystemParamsCompartment _compartment_;
    private SystemParamsCompartment _nextCompartment_;

    SystemParams() {
//        _state_ = this::_sBegin_;
        // _stateContext_ = new StateContext(this::_sBegin_);
    }

    SystemParams(String stateMsg, String enterMsg) {

        // Create and intialize start state compartment.

//        this._state_=this::_sBegin_;
        this._compartment_=new SystemParamsCompartment(this._state_);
        this._nextCompartment_=null;
        this._compartment_.stateArgs.put("stateMsg", stateMsg);
        this._compartment_.enterArgs.put("enterMsg", enterMsg);

        // Initialize domain

        // Send system start event
        FrameEvent frameEvent = new FrameEvent(">", this._compartment_.enterArgs);
        this._mux_(frameEvent);
    }

    //====================== Multiplexer ====================//

    private void _mux_(FrameEvent e){
    			
        if(this._compartment_.state==SystemParamState.BEGIN.getValue()) {
            this._sBegin_(e);
        }

        if(this._nextCompartment_ != null) {
            SystemParamsCompartment nextCompartment = this._nextCompartment_;
            this._nextCompartment_ = null;
            if(nextCompartment._forwardEvent != null &&
              nextCompartment._forwardEvent._message == ">") {
                this._mux_(new FrameEvent("<", this._compartment_.exitArgs));
                this._compartment_ = nextCompartment;
                this._mux_(nextCompartment._forwardEvent);
            } else {
                this._doTransition_(nextCompartment);
                if (nextCompartment._forwardEvent != null) {
                    this._mux_(nextCompartment._forwardEvent);
                }
            }
            nextCompartment._forwardEvent = null;
        }
    }
    
    //===================== Machine Block ===================//
    
    
    private void _sBegin_(FrameEvent e) {
        if (e._message == ">") {
            print_do(((String) this._compartment_.stateArgs.get("stateMsg")) + " " + ((String) e._parameters.get("enterMsg")));
            return;
        }
    }

    //===================== Actions Block ===================//
    
    protected void print_do(String msg) { throw new UnsupportedOperationException(); }
    
    
    //=============== Machinery and Mechanisms ==============//
    
    private int _state_;
    // private StateContext _stateContext_;

    private void _transition_(SystemParamsCompartment compartment) {
        this._nextCompartment_=compartment;
    }

    private void _doTransition_(SystemParamsCompartment nextCompartment) {
        this._mux_(new FrameEvent("<", this._compartment_.exitArgs));
        this._compartment_ = _nextCompartment_;
        this._mux_(new FrameEvent(">", this._compartment_.enterArgs));
    } 
    
}

//====================Compartment====================//

class SystemParamsCompartment {

    int state;

    SystemParamsCompartment(int state) {
        this.state=state;
    }

    HashMap<String,Object> stateArgs = new HashMap<String,Object>();
	HashMap<String,Object> stateVars = new HashMap<String,Object>();
	HashMap<String,Object> enterArgs = new HashMap<String,Object>();
    HashMap<String,Object> exitArgs = new HashMap<String,Object>();

    FrameEvent _forwardEvent = new FrameEvent();
}

class SystemParamsController extends SystemParams {

    String stateMsg;
    String enterMsg;

    public SystemParamsController() {}

    public SystemParamsController(String stateMsg, String enterMsg) {
        super(stateMsg, enterMsg);
    }

    protected void print_do(String msg) {
        System.out.println(msg);
    }
}

File: csharp/SystemParams/machine.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

using SystemParams;
#nullable disable
namespace csharp.systemparams
{

class SystemParams
{

    private SystemParamsCompartment _compartment_;
    private SystemParamsCompartment _nextCompartment_;


    public SystemParams(string stateMsg, string enterMsg)
    {


        // Create and intialize start state compartment.


        this._state_ = (int)SystemParamsState.BEGIN;
        this._compartment_ = new SystemParamsCompartment(this._state_);
        this._nextCompartment_ = null;
        this._compartment_.StateArgs["stateMsg"] = stateMsg;
        this._compartment_.EnterArgs["enterMsg"] = enterMsg;


        // Send system start event
        FrameEvent frameEvent = new FrameEvent(">", this._compartment_.EnterArgs);
        this._mux_(frameEvent);

    }

    // states enum
    private enum SystemParamsState
    {

        BEGIN = 0
    }
    //====================== Multiplexer ====================//

    private void _mux_(FrameEvent e)
    {
        switch (this._compartment_.state)
        {
            case (int)SystemParamsState.BEGIN:
                this._sBegin_(e);
                break;
        }

        if( this._nextCompartment_ != null)
        {
            var nextCompartment = this._nextCompartment_;
            this._nextCompartment_ = null;
            if (nextCompartment._forwardEvent != null &&
               nextCompartment._forwardEvent._message == ">")
            {
                this._mux_(new FrameEvent( "<", this._compartment_.ExitArgs));
                this._compartment_ = nextCompartment;
                this._mux_(nextCompartment._forwardEvent);
            }
            else
            {
                this._doTransition_(nextCompartment);
                if (nextCompartment._forwardEvent != null)
                {
                    this._mux_(nextCompartment._forwardEvent);
                }
            }
            nextCompartment._forwardEvent = null;
        }
    }


    //===================== Machine Block ===================//

    SystemParamsCompartment compartment;


    private void _sBegin_(FrameEvent e)
    {
        if (e._message == ">")
        {
            this.print_do(((string)this._compartment_.StateArgs["stateMsg"]) + " " + ((string)e._parameters["enterMsg"]));
            return;
        }
    }

    //===================== Actions Block ===================//

    public void print_do(string msg)
    {

        Console.WriteLine(msg);

    }

    // Unimplemented Actions



    //=============== Machinery and Mechanisms ==============//

    private int _state_;

    private void _transition_(SystemParamsCompartment compartment)
    {
        this._nextCompartment_ = compartment;
    }

    private void _doTransition_(SystemParamsCompartment nextCompartment)
    {
        this._mux_(new FrameEvent("<", this._compartment_.ExitArgs));
        this._compartment_ = nextCompartment;
        this._mux_(new FrameEvent(">", this._compartment_.EnterArgs));
    }

}

//=============== Compartment ==============//

class SystemParamsCompartment
{

    public int state;

    public SystemParamsCompartment(int state)
    {
        this.state = state;
    }

    public Dictionary<string, object> StateArgs { get; set; } = new Dictionary<string, object>();
    public Dictionary<string, object> StateVars { get; set; } = new Dictionary<string, object>();
    public Dictionary<string, object> EnterArgs { get; set; } = new Dictionary<string, object>();
    public Dictionary<string, object> ExitArgs { get; set; } = new Dictionary<string, object>();
    public FrameEvent _forwardEvent = new FrameEvent();
}

class SystemParamsController : SystemParams
{
        public SystemParamsController(string stateMsg, string enterMsg) : base(stateMsg, enterMsg)
        {
        }
}

}